### PR TITLE
Chore: Mark up User Profile page for translation

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Annotates the components used on the User Profile page for translation.

Adds [psuedolocalisation](https://lingui.js.org/guides/pseudolocalization.html) to verify work without actually translating it.

**Which issue(s) this PR fixes**:

Relates to #41610 

**Special notes for your reviewer**:

First two commits will be removed as they're in another PR.

Questions raised from doing this:

 - Naming convention for translation IDs
 - When to use `<Trans />` vs `t({ })`

### CODE REVIEW

File: public/app/core/components/SharedPreferences/SharedPreferences.tsx [hunk 1]
```diff
@@ -163,30 +177,40 @@ export class SharedPreferences extends PureComponent<Props, State> {
...
+              >
                 <WeekStartPicker
                   value={weekStart}
                   onChange={this.onWeekStartChanged}
                   inputId={'shared-preferences-week-start-picker'}
                 />
               </Field>
+
               <div className="gf-form-button-row">
                 <Button variant="primary" aria-label="User preferences save button">
```
Comment (joshhunt) [inline public/app/core/components/SharedPreferences/SharedPreferences.tsx:212]: TODO: Change this to data-testid e2e selector

File: public/app/features/admin/state/actions.ts [hunk 2]
```diff
@@ -142,14 +143,21 @@ export function loadUserSessions(userId: number): ThunkResult<void> {
...
     tokens.reverse();
-    const sessions = tokens.map((session: UserSession) => {
+
+    const sessions = tokens.map((session) => {
       return {
         id: session.id,
         isActive: session.isActive,
         seenAt: dateTimeFormatTimeAgo(session.seenAt),
-        createdAt: dateTimeFormat(session.createdAt, { format: 'MMMM DD, YYYY' }),
+        createdAt: new Date(session.createdAt),
```
Comment (joshhunt) [inline public/app/features/admin/state/actions.ts:160]: TODO: I made this change to the _admin_ model, but I havent updated the component used on the admin page.

File: public/app/features/profile/state/reducers.ts [hunk 3]
```diff
@@ -78,7 +78,7 @@ export const slice = createSlice({
         id: session.id,
         isActive: session.isActive,
         seenAt: dateTimeFormatTimeAgo(session.seenAt),
-        createdAt: dateTimeFormat(session.createdAt, { format: 'MMMM DD, YYYY' }),
+        createdAt: new Date(session.createdAt),
```
Comment (joshhunt) [inline public/app/features/profile/state/reducers.ts:81]: This is possibly the one contentious part of this PR that I want feedback/extra eyes on. 

To format the date correctly, with respect to preferred localisation, in the view, we need access to the createdAt as an actual Date object. Besides, I think it is "less correct" to store the date as a formatted string in the store anyway.

File: public/app/features/profile/UserProfileEditForm.tsx [hunk 4]
```diff
@@ -22,37 +23,57 @@ export const UserProfileEditForm: FC<Props> = ({ user, isSavingUser, updateProfi
     <Form onSubmit={onSubmitProfileUpdate} validateOn="onBlur">
       {({ register, errors }) => {
         return (
-          <FieldSet label={<Trans id="edit-user-profile.title">Edit profile</Trans>}>
-            <Field label="Name" invalid={!!errors.name} error="Name is required" disabled={disableLoginForm}>
+          <FieldSet label={<Trans id="user-profile.title">Edit profile</Trans>}>
+            <Field
+              label={t({ id: 'user-profile.fields.name-label', message: 'Name' })}
```
Comment (joshhunt) [inline public/app/features/profile/UserProfileEditForm.tsx:28]: I would much prefer to do 

```typescript
            <Field
              label={<Trans id="user-profile.fields.name-label">Name</Trans>}
```

However, the `label` prop here treats it different if you pass in ReactNode vs string.

I would prefer to use `<Trans />` everywhere except when a string is strictly required (like `placeholder` on inputs) where we use `t()` instead. I find `<Trans />` easier to read than all the jumble of brackets from `label={t({})}`.

Comment (ashharrison90) [reply public/app/features/profile/UserProfileEditForm.tsx:28]: i don't mind either syntax so much tbh. the only thing i'm not super keen on is the function being called `t`. i guess it's for brevity, but the component isn't called `<T />` so idk why they made the function `t()`? i know we can change it on import, but that relies on every developer remembering to do that 😬

Comment (joshhunt) [reply public/app/features/profile/UserProfileEditForm.tsx:28]: I see what you mean, and while yes "one letter variables" are not preferred, I personally don't have much of a problem with a very specific known one letter function.

Either way, I think it we could automate and/or lint whatever we do here.

**Review by kaydelaney — APPROVED** [review]
Looks good!

File: public/app/features/profile/state/reducers.ts (hunk 3 above)
Comment (ashharrison90) [reply public/app/features/profile/state/reducers.ts:81]: mmmmmm i agree this is a little contentious 😛 

storing objects in a redux store is a little funky. i think you're only supposed to store JSON serializable stuff in there? could we store either an ISO timestamp or the UNIX timestamp and creating the Date object at runtime? 🤔

Comment (joshhunt) [reply public/app/features/profile/state/reducers.ts]: Good shout. I didn't know storing non-json serialisable values in Redux was not recommended!

**Review by ashharrison90 — APPROVED** [review]
looks great! 🚀

**Pull request merged**